src/IO/H5/CombineH5.hpp

```diff
@@ -6,8 +6,44 @@
 #include <string>
 #include <vector>
 
+#include "IO/Logging/Verbosity.hpp"
+
 namespace h5 {
 void combine_h5_vol(const std::vector<std::string>& file_names,
                     const std::string& subfile_name, const std::string& output,
                     bool check_src = true);
+
+/*!
+ * \brief Combine the `h5::Dat` subfiles of multiple `h5::H5File`s into a single
+ * H5 file.
+ *
+ * \details If there are overlapping times, the "latest" one is always used;
+ * meaning if you have data in `File1.h5` and `File2.h5` and if the first time
+ * in `File2.h5` is before some times in `File1.h5`, those times in `File1.h5`
+ * will be discarded and won't appear in the combined H5 file.
```

Comment:
I think the constraint is quite strong and should be explicit: the files past must be in increasing time order.